ui-v2/src/routes/work-pools/work-pool.$workPoolName.tsx

```diff
@@ -1,9 +1,22 @@
+import { buildGetWorkPoolQuery } from "@/api/work-pools/work-pools";
 import { createFileRoute } from "@tanstack/react-router";
 
 export const Route = createFileRoute("/work-pools/work-pool/$workPoolName")({
 	component: RouteComponent,
+	loader: async ({ context, params }) => {
+		const { workPoolName } = params;
+
+		const workPool = await context.queryClient.ensureQueryData(
+			buildGetWorkPoolQuery(workPoolName),
+		);
+
+		return { workPool };
+	},
+	wrapInSuspense: true,
```

Comment:
```suggestion
	loader: async ({ context, params }) => {
		const { workPoolName } = params;

		 return context.queryClient.ensureQueryData(buildGetWorkPoolQuery(workPoolName));
	},
	wrapInSuspense: true,
```
And below, you would subscribe to the fetched data using `useSuspenseQuery(buildGetWorkPoolQuery(workPoolName))`  (be sure to fetch workPoolName from the params again)